### PR TITLE
Observations in Localization query: use caching

### PR DESCRIPTION
**Pull request by Theodlz**
In this PR, we add 2 features:
- Use cached instrument and localization overlap if available when fetching the observations of an instrument in a localization. In a following PR, we can imagine caching it here as well, but that is not a priority as common values of cumprob like 0.90 and 0.95 will be cached by other features already.
- Specify which instrument to get the observations for when using the GCN Summary feature (optional, all if not specified)

### CODE REVIEW

File: skyportal/handlers/api/gcn.py [hunk 1]
```diff
@@ -1918,8 +1918,9 @@ def add_gcn_summary(
     show_observations,
     no_text,
     photometry_in_window,
+    stats_method,
```
Comment (mcoughlin) [inline skyportal/handlers/api/gcn.py:1921]: Maybe these should have defaults?

File: skyportal/handlers/api/observation.py [hunk 2]
```diff
@@ -57,6 +58,14 @@
 
 Session = scoped_session(sessionmaker())
 
+cache_dir = "cache/localization_instrument_queries"
+cache = Cache(
+    cache_dir=cache_dir,
+    max_items=cfg.get("misc.max_items_in_localization_instrument_query_cache", 100),
+    max_age=cfg.get("misc.minutes_to_keep_localization_instrument_query_cache", 24 * 60)
```
Comment (mcoughlin) [inline skyportal/handlers/api/observation.py:65]: Just maybe comment that this is 1 day?

File: static/js/components/GcnSummary.jsx [hunk 3]
```diff
@@ -174,6 +177,36 @@ const GcnSummary = ({ dateobs }) => {
     label: `${user.first_name} ${user.last_name}`,
   }));
 
+  let sortedInstrumentList = [...instrumentList];
+  sortedInstrumentList.sort((i1, i2) => {
+    // if the instrument is ZTF, put it first
+    if (i1.name === "ZTF") {
```
Comment (mcoughlin) [inline static/js/components/GcnSummary.jsx:183]: Does ZTF need hard-coding here?

**Review by mcoughlin — APPROVED** [review]
Small suggestions.

File: static/js/components/GcnSummary.jsx (hunk 3 above)
Comment (Theodlz) [reply static/js/components/GcnSummary.jsx:183]: Will delete

File: skyportal/handlers/api/observation.py (hunk 2 above)
Comment (Theodlz) [reply skyportal/handlers/api/observation.py:65]: ok!

File: skyportal/handlers/api/gcn.py (hunk 1 above)
Comment (Theodlz) [reply skyportal/handlers/api/gcn.py:1921]: Agreed, will add

**Review by mcoughlin — APPROVED** [review]
LGTM

**Pull request merged**